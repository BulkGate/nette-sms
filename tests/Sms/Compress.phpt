<?php

/**
 * Test: Nette\Object magic @methods and types.
 */

namespace Test;

use Nette;
use Nette\Sms\Compress;
use Tester\Assert;

require __DIR__ . '/../bootstrap.php';
require __DIR__ . '/../../src/Sms/Compress.php';



Assert::same('H4sIAAAAAAACC0u0MrKqLrYytlJKTEpWss60MjQyhpII0VoASinDoSYAAAA=', Compress::compress(['abc' => 123, '123' => 'abc']));

Assert::same('H4sIAAAAAAAEC0u0MrKqLrYytlJKTEpWss60MjQyhpII0VoASinDoSYAAAA=', Compress::compress(['abc' => 123, '123' => 'abc'], 1));

Assert::same(['abc' => 123, '123' => 'abc'], Compress::decompress('H4sIAAAAAAAEC0u0MrKqLrYytlJKTEpWss60MjQyhpII0VoASinDoSYAAAA='));

Assert::same(['abc' => 123, '123' => 'abc'], Compress::decompress(Compress::compress(['abc' => 123, '123' => 'abc'])));

$json = json_encode([
	'status' => '-1',
	'type' => 'error',
	'message' => 'Unknown action',
	'compress' => false,
	'data' => [
		123456789,
		123456789,
		123456789,
		123456789,
		123456789,
		123456789,
		123456789,
		123456789,
		123456789,
		123456789,
		123456789,
		123456789,
		123456789,
		123456789,
		123456789,
	],
]);

Assert::true(strlen($json) > strlen(Compress::compress($json)));
