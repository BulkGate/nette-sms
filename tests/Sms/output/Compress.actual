array('abc' => 123, 123 => 'abc')
