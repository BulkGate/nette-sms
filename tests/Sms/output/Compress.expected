array('ab1c' => 123, 123 => 'abc')
